styles/layers.h

```diff
@@ -4,6 +4,15 @@
 #include "alpha.h"
 #include "../functions/int.h"
 #include "colors.h"
+#include <type_traits>
+#include <utility>
+
+// Identify “solid” colors by the return type of getColor()
+namespace layers_detail {
```

Comment:
This seems like it could be useful in lots of places, so maybe call it color_details and put it in common/color.h instead?

Reply:
Done.